catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -0,0 +1,507 @@
+    private final FilterAdapter filterAdapter;
+
+    private final DynamicSchemaResolver resolver;
+
+    public SolrMetacardClientImpl(SolrClient client, FilterAdapter catalogFilterAdapter,
+            SolrFilterDelegateFactory solrFilterDelegateFactory,
+            DynamicSchemaResolver dynamicSchemaResolver) {
+        this.client = client;
+        filterDelegateFactory = solrFilterDelegateFactory;
+        filterAdapter = catalogFilterAdapter;
+        resolver = dynamicSchemaResolver;
+    }
+
+    public SolrClient getClient() {
+        return client;
+    }
+
+    @Override
+    public SourceResponse query(QueryRequest request) throws UnsupportedQueryException {
+        if (request == null || request.getQuery() == null) {
+            return new QueryResponseImpl(request, new ArrayList<Result>(), true, 0L);
+        }
+
+        SolrQuery query = getSolrQuery(request, filterDelegateFactory.newInstance(resolver));
+
+        long totalHits;
+        List<Result> results = new ArrayList<>();
+        try {
+            QueryResponse solrResponse = client.query(query, SolrRequest.METHOD.POST);
+            totalHits = solrResponse.getResults()
+                    .getNumFound();
+            SolrDocumentList docs = solrResponse.getResults();
+
+            for (SolrDocument doc : docs) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("SOLR DOC: {}",
+                            doc.getFieldValue(Metacard.ID + SchemaFields.TEXT_SUFFIX));
+                }
+                ResultImpl tmpResult;
+                try {
+                    tmpResult = createResult(doc);
+                    // TODO: register metacard type???
+                } catch (MetacardCreationException e) {
+                    throw new UnsupportedQueryException("Could not create metacard(s).", e);
+                }
+
+                results.add(tmpResult);
+            }
+
+        } catch (SolrServerException | IOException | SolrException e) {
+            throw new UnsupportedQueryException("Could not complete solr query.", e);
+        }
+
+        SourceResponseImpl sourceResponseImpl = new SourceResponseImpl(request, results);
+
+        /* Total Count */
+        sourceResponseImpl.setHits(totalHits);
+
+        return sourceResponseImpl;
+    }
+
+    @Override
+    public List<Metacard> query(String queryString) throws UnsupportedQueryException {
+        SolrQuery query = new SolrQuery();
+        query.setQuery(queryString);
+        try {
+            QueryResponse solrResponse = client.query(query, SolrRequest.METHOD.POST);
+            SolrDocumentList docs = solrResponse.getResults();
+
+            List<Metacard> results = new ArrayList<>();
+            for (SolrDocument doc : docs) {
+                try {
+                    results.add(createMetacard(doc));
+                } catch (MetacardCreationException e) {
+                    throw new UnsupportedQueryException("Could not create metacard(s).", e);
```

Comment:
I agree that it is odd but the `Source` interface does not give us much else to throw without using a runtime exception.